src/Repository/PageRepository.php

```diff
@@ -25,12 +25,28 @@ public function all(array $search = [], ?int $skip = null, ?int $limit = null, a
         return parent::all($search, $skip, $limit, $columns, $orderDirection, $orderColumn);
     }
 
-
     /**
      * @param string $slug
      * @return Page
      */
-    public function getBySlug(string $slug) {
-        return $this->model->newQuery()->where('slug', $slug)->first();
+    public function getBySlug(string $slug): Page {
+        return $this->model->newQuery()->where('slug', $slug)->firstOrNew();
```

Comment:
I think that you should probably have separate function that returns first or new because `getBySlug` sounds like it should return null if the page does not exist, and not a new Model

Reply:
But the method has Page return value, so it should be clear what's the return type. Please propose an alternative method name if you still would like it to be different.

Reply:
But what is usecase of this function, like when would you really want to get an empty Page object when you are trying to retrieve something by Slug? Then again, it enables us to use ->exists() on the result returned from this function.

Either leave it as is, or maybe rename it `findBySlugOrNew()`?

Reply:
Thanks, ->exists() on the resulting model is a valid use-case for that function indeed.